lib/lightning_web/live/components/project_deletion_modal.ex

```diff
@@ -0,0 +1,130 @@
+defmodule LightningWeb.Components.ProjectDeletionModal do
+  @moduledoc false
+  use LightningWeb, :component
+
+  use Phoenix.LiveComponent
+
+  alias Lightning.Projects
+  alias Lightning.Projects.Project
+
+  @impl true
+  def update(%{project: project} = assigns, socket) do
+    {:ok,
+     socket
+     |> assign(
+       delete_now?: !is_nil(project.scheduled_deletion),
+       scheduled_deletion_changeset: Projects.change_scheduled_deletion(project)
+     )
+     |> assign(assigns)}
+  end
+
+  @impl true
+  def handle_event(
+        "validate",
+        %{"project" => project_params},
+        socket
+      ) do
+    changeset =
+      socket.assigns.project
+      |> Projects.change_scheduled_deletion(project_params)
+      |> Map.put(:action, :validate_scheduled_deletion)
```

Comment:
Just curious, why `:validation_scheduled_deletion`, normally we put `:validate` because of LiveView forms, do the errors appear correctly on the form if you don't enter the matching project name?